### PR TITLE
Ingress removal storage svc

### PR DESCRIPTION
**Pull request by IsaiasA1**
# Description
 This PR removes the storage-service ingress from the RPM, adds storage-service handlers to the proxy-server, updates karavictl to connect to the proxy-server instead of the storage-service, and adds telemetry updates.

Proxy-Server changes:

New service handler at internal/proxy/service_handler.go to handle these paths for karavictl:

POST /proxy/storage/ (create storage, storage info is in request body)
PATCH /proxy/storage/ (update storage, storage info is in request body)
GET /proxy/storage/ (list or get storage, storage name is in query parameter)
PATCH /proxy/storage/ (storage info is in request body)
The handlers for each path will parse the request and make the appropriate grpc request to the storage service.

Adds grpc connection interceptors which will automatically create tracing spans for grpc client requests.

New http middleware called TelemetryMW. This will log the time taken for the next handler to complete and record the error in the span. The telemetry will happen only if the next handler is the type HandlerWithError. This handler returns an error specifically for recording in the span.

type HandlerWithError func(w http.ResponseWriter, r *http.Request) error

karavictl changes:

karavictl commands that previously connected to the storage-service via grpc now connect to the proxy-server via https
New file cmd/karavictl/cmd/api/api.go which is a generic http client inspired by https://github.com/dell/goscaleio/blob/main/api/api.go
karavictl will build this generic http client and execute the appropriate request
Storage-service changes:

Adds grpc server interceptors which will automatically create tracing spans for incoming requests.
Adds telemetry middleware at internal/storage-service/middleware/storage_telemetry.go to log the time taken for the request, log request info, and set the request info in the span

Most if not all code is similar to Tenant-service changes made by Aaron Tye 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/725|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Powerflex E2E Test, and manual test (Containerized, RPM, and CSM-Operator)

### CODE REVIEW

File: cmd/karavictl/cmd/storage_create_test.go [hunk 1]
```diff
@@ -335,3 +339,107 @@ func Test_readPassword(t *testing.T) {
...
+		}
+		var gotOutput bytes.Buffer
+
+		cmd := NewRootCmd()
+		cmd.SetOutput(&gotOutput)
+		cmd.SetArgs([]string{"storage", "create", "--addr", "https://storage-service.com", "--endpoint", "https://0.0.0.0:443", "--system-id", "testing123", "--type", "powerflex", "--user", "admin", "--password", "password", "--insecure", "--array-insecure"})
+		cmd.Execute()
+
+		if !gotCalled {
+			t.Error("expected DeleteTenant to be called, but it wasn't")
```
Comment (xuluna) [inline cmd/karavictl/cmd/storage_create_test.go:374]: Is it DeleteTenant to be called? Also the gotCalled was set to true anyways from the beginning?

Comment (IsaiasA1) [reply cmd/karavictl/cmd/storage_create_test.go]: fixed, also noticed this on other karavictl test and updated the error logs

**Review by alikdell — APPROVED** [review]

File: internal/storage-service/middleware/storage_telemetry_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package middleware
```
Comment (xuluna) [inline internal/storage-service/middleware/storage_telemetry_test.go:1]: Copy Right headers.

Comment (IsaiasA1) [reply internal/storage-service/middleware/storage_telemetry_test.go:13]: added

**Review by alikdell — APPROVED** [review]

File: cmd/storage-service/main.go [hunk 3]
```diff
@@ -41,10 +52,42 @@ const (
 	concurrentPowerFlexRequests = "CONCURRENT_POWERFLEX_REQUESTS"
 )
 
+var (
+	cfg Config
+)
+
+// Config is the configuration details on the tenant-service
```
[the remark this answers is not in the document]
Comment (IsaiasA1) [reply cmd/storage-service/main.go]: fixed

File: internal/storage-service/mocks/storage_server.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// Copyright © 2021-2023 Dell Inc., or its subsidiaries. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (IsaiasA1) [reply internal/storage-service/mocks/storage_server.go]: fixed

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by alikdell — DISMISSED** [review]

**Comment by IsaiasA1** [comment]
run e2e test

**Comment by IsaiasA1** [comment]
run e2e test

**Review by xuluna — APPROVED** [review]

**Review by shaynafinocchiaro — APPROVED** [review]

**Comment by IsaiasA1** [comment]
run e2e test

**Pull request merged**